lib/cn/csched_sp3_work.c

```diff
@@ -871,8 +1040,14 @@ sp3_work(
     w->cw_t0_enqueue = get_time_ns();
 
     /* Ensure concurrent root spills complete in order */
-    if (w->cw_action == CN_ACTION_SPILL)
+    if (w->cw_action == CN_ACTION_SPILL) {
         w->cw_sgen = ++tree->ct_sgen;
+    } else if (w->cw_action == CN_ACTION_JOIN) {
+
+        /* TODO: Should we set the work ID of each kvset in cw_join?
+         */
+        w->cw_join = list_prev_entry(tn, tn_link);
+    }
```

Comment:
@nabeelmmd, either do this or delete the comment.  I think the work ID stuff is primarily only needed to run concurrent jobs on the same node (e.g., spills on the root node), but we don't allow that for any leaf node...

Reply:
I think it's better to set the work ID for the marked kvsets in cw_join. I presume we will need that for zspill?

Reply:
Yes, zpill will need to do that to allow concurrent root spills.